lib/internal/test_runner/test.js

```diff
@@ -136,8 +139,11 @@ class Test extends AsyncResource {
       this.concurrency = concurrency;
     }
 
-    if (isUint32(timeout)) {
-      this.timeout = timeout;
+    if (timeout != null && timeout !== Infinity) {
+      validateNumber(timeout, 'options.timeout', 0);
+      if (timeout < TIMEOUT_MAX) {
```

Comment:
I realize that this is a ridiculously long timeout (roughly 25 days) for most purposes, but should we at least emit a warning otherwise? And/or set the timeout to `TIMEOUT_MAX`? Or throw?

Reply:
Ah yes, I didn't realize it was that long. I think that throwing an ERR_OUT_OF_RANGE is the most helpful thing to do here, but no strong feeling, setting overflow values as `TIMEOUT_MAX` and emitting a warning also sounds fine.